deploy_aws.sh

```diff
@@ -0,0 +1,116 @@
+  echo "==== installing homebrew ===="
+fi
+
+# Install aws cli
+if ( which aws > /dev/null ) 
+then
+  echo -e "\n==== awscli already installed ===="
+else 
+  brew install awscli
+  echo "\n==== installing awscli ===="
+fi
+
+# Install jq
+if ( which jq > /dev/null ) 
+then
+  echo -e "\n==== jq already installed ===="
+else 
+  brew install jq
+  echo "\n==== installing jq ===="
+fi
+
+# Check if an aws key-pair is present
+#  pipe the key pair to jq and write that key pair to a file named id_rsa 
+if ( aws ec2 describe-key-pairs --key-name aws_rsa_key_pair --out text )
+then
+  echo -e "==== key pair present ===="
+else
+  echo -e "==== creating key pair ===="
+  aws ec2 create-key-pair --key-name aws_rsa_key_pair --query 'KeyMaterial' --output text > aws_rsa_key_pair.pem 
+  chmod 0600 aws_rsa_key_pair.pem 
+fi
+
+# Check if the security group exists (if it is -z zero length)
+security_group_id=$(aws ec2 describe-security-groups --group-names ssh_http --region $AWS_DEFAULT_REGION --output json | jq -r '.SecurityGroups[0].GroupId')
+if [ -z "$security_group_id" ]
+then
+  # Create the security group
+  security_group_id=$(aws ec2 create-security-group --group-name ssh_http --description "Allow SSH and HTTP" --region $AWS_DEFAULT_REGION --output json | jq -r '.GroupId')
+  echo -e "==== security group created ===="
+
+  # Add an ingress rule to the security group to allow incoming SSH connections using the aws ec2 authorize-security-group-ingress command
+  # The --group-id flag specifies the ID of the security group to modify
+  # The --protocol flag specifies the protocol (TCP in this case)
+  # The --port flag specifies the port number (22 for SSH)
+  # The --cidr flag specifies the IP address range that is allowed to make the connection
+  # The --region flag specifies the region in which the security group was created
+  aws ec2 authorize-security-group-ingress --group-id $security_group_id --protocol tcp --port 22 --cidr 0.0.0.0/0 --region $AWS_DEFAULT_REGION
+  aws ec2 authorize-security-group-ingress --group-id $security_group_id --protocol tcp --port 80 --cidr 0.0.0.0/0 --region $AWS_DEFAULT_REGION
+
+  # Add an egress rule to the security group to allow all outbound traffic using the aws ec2 authorize-security-group-egress command
+  # The --group-id flag specifies the ID of the security group to modify
+  # The --protocol flag specifies the protocol (-1 for all protocols)
+  # The --port flag specifies the port number (0 for all ports)
+  # The --cidr flag specifies the IP address range that is allowed to receive the outbound traffic
+  # The --region flag specifies the region in which the security group was created
+  aws ec2 authorize-security-group-egress --group-id $security_group_id --protocol -1 --port 0 --cidr 0.0.0.0/0 --region $AWS_DEFAULT_REGION
+
+# If the security group already exists, print a message
+else
+  echo -e "==== security group already exists ===="
+fi
+
+<<exists
```

Comment:
Do we need any of this ami code if we're just hardcoding it into the run-instances line?